label_studio/io_storages/gcs/utils.py

```diff
@@ -30,6 +31,21 @@ class ConvertBlobTo(Enum):
         JSON_DICT = 3
         BASE64 = 4
 
+    @classmethod
+    @lru_cache(maxsize=1)
```

Comment:
It strikes me that leaving the gcs bucket around with an infinite TTL might lead to unintended behavior - suppose for example that the bucket is deleted in GCS, this function would continue to return the bucket, even though an actual call to `get_client` would no longer return it. An alternative approach would be to use a request-scoped cache for the bucket, which I think would provide an equivalent speedup in the example from the LEAP-19 ticket (since it's a single request trace). See for example: https://pypi.org/project/django-request-cache/ which I think could be useful in other settings as well.

Reply:
I don't want to add 2 third party middlewares for this. 

Cache has size of 1 element, so `unintended behavior` will be quite rare, and even with request based cache it is still possible, bucket can be deleted after we initialized object. 

I can add ttl to this cache, do you think it's safe enough?

Reply:
A TTL would indeed get us most of the way there. I've been thinking for a while that we could benefit from having this package in the build if we don't already: https://cachetools.readthedocs.io/en/latest/#cachetools.TTLCache